libraries/resource_providers/api500/c7000/volume_provider.rb

```diff
@@ -0,0 +1,107 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API500
+    module C7000
+      # Volume API500 C7000 provider
+      class VolumeProvider < API300::C7000::VolumeProvider
+        def load_resource_with_associated_resources
+          if @new_resource.volume_template
+            load_volume_template
+          else
+            validate_required_properties(:storage_system, :storage_pool)
+            load_storage_pool
+            load_snapshot_pool if @new_resource.snapshot_pool
+          end
+          set_properties unless @item.exists?
+        end
+
+        def load_storage_system
+          data = {
+            hostname: @new_resource.storage_system,
+            name: @new_resource.storage_system
+          }
+          @storage_system = load_resource(:StorageSystem, data)
+        end
+
+        def load_storage_pool
+          load_storage_system
+          super
+        end
+
+        def load_volume_template
+          template = resource_named(:VolumeTemplate).new(@item.client, name: @new_resource.volume_template)
+          @item.set_storage_volume_template(template)
+          @item['storagePoolUri'] = template['storagePoolUri']
+          @item['snapshotPoolUri'] = template['storagePoolUri'] if template['family'] == 'StoreServ'
```

Comment:
Must item[**'snapshotPoolUri'**] receive the value from template[**'storagePoolUri'**] ? Or would be ` @item['snapshotPoolUri'] = template['snapshotPoolUri']`?

Reply:
The template doesn't have this attribute, but there is one ```template['properties']['snapshotPool']['default']``` , I will use it. This attribute has the snapshotUri assigned in creating the template volume.